pkg/gui/controllers/helpers/refresh_helper.go

```diff
@@ -249,6 +249,32 @@ func (self *RefreshHelper) refreshCommits() {
 	wg.Wait()
 }
 
+func (self *RefreshHelper) determineCheckoutBranchName() string {
```

Comment:
This function doesn't depend on anything outside the git_commands package so I'd move it there

Reply:
It needs several different git command objects though, so `self` would have to be `*GitCommands`, I guess. We don't have precedence for that yet. Should this go into a new file?

Reply:
I can't see why not

Reply:
It can't go in the `git_commands` package, because `GitCommands` is defined in `commands`, not `git_commands`. Unless we import `commands` from this new file in `git_commands`, which doesn't seem right to me.

I'm happy to move it somewhere if you make a more concrete suggestion where it should go, I can't figure this out myself. Personally I'd just keep it as a local helper in refresh_helper.go.